src/index.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useState, useRef } from 'react';
+
+type IWindowProps = {
+  url: string;
+  title: string;
+  width: number;
+  height: number;
+};
+
+type IPopupProps = IWindowProps & {
+  onClose: () => void;
+  onCode: (code: string, params: URLSearchParams) => void;
+  children: React.ReactNode;
+};
+
+const createPopup = ({
+  url, title, height, width,
+}: IWindowProps) => {
+  const left = window.screenX + (window.outerWidth - width) / 2;
+  const top = window.screenY + (window.outerHeight - height) / 2.5;
+  const externalPopup = window.open(
+    url,
+    title,
+    `width=${width},height=${height},left=${left},top=${top}`,
+  );
+  return externalPopup;
+};
+
+const OauthPopup: React.FC<IPopupProps> = ({
+  title = '',
+  width = 500,
+  height = 500,
+  url,
+  children,
+  onCode,
+  onClose,
+}: IPopupProps) => {
+  const [externalWindow, setExternalWindow] = useState<Window | null>();
+  const intervalRef = useRef<number>();
+
+  const clearTimer = () => {
+    window.clearInterval(intervalRef.current);
+  };
+
+  const onContainerClick = () => {
+    setExternalWindow(createPopup({
+      url, title, width, height,
+    }));
+  };
+
+  useEffect(() => {
+    if (externalWindow) {
+      intervalRef.current = window.setInterval(() => {
```

Comment:
Are you intending to open a window every 700ms?

Reply:
Every 700ms it checks if the oauth popup was successfully submitted and if a redirect happened. Does that make sense?